lib/src/component_declaration/component_base.dart

```diff
@@ -275,23 +275,27 @@ typedef PropsModifier(Map props);
 abstract class UiProps
     extends Object with MapViewMixin, PropsMapViewMixin, ReactPropsMixin, UbiquitousDomPropsMixin, CssClassPropsMixin
     implements Map {
+
   /// Adds an arbitrary prop key-value pair.
-  void addProp(propKey, value) {
+  /// IF [shouldAdd] is true, otherwise, does nothing.
```

Comment:
#nit IF should not be captialized (I realized this was carried over from `add`/`blacklist`, but that seems like a typo).

Also, this line is a sentence fragment, and should probably be combined with the previous sentence.
```dart
/// Adds an arbitrary prop key-value pair if [shouldAdd] is true, otherwise, does nothing.
```